### DISCUSSION TITLE
Remove duplicates not working?

### DISCUSSION THREAD

**Discussion by lsmithg12** [CrowdStrike/psfalcon#235] [Q&A] [answered]
Hello! Thanks for all the work you do with PSFalcon.

I am looking to remove duplicates from devices that were reimaged. Finding the duplicates works and creates the .csv but when I run the script that you have built that also hides them, the script runs without error, but the devices are never removed in the CS Console. 

Is there anything that I could be missing or doing wrong? There are no errors and the script completes. I checked my API access and have enabled read/write for everything.

Thanks for any help!

**Comment by bk-cs** [comment, marked as answer]
Can you elaborate on what steps you're following? Script content, commands run, etc.

The general process looks something like this:
```
# Export duplicates into a CSV
Find-FalconDuplicate | Export-Csv .\possible_duplicates.csv
# If the CSV looks valid, import results and hide them
Import-Csv .\possible_duplicates.csv | Invoke-FalconHostAction -Name hide_host
```

**Reply by lsmithg12** [reply]
Definitely! I am using your code from the Basic Scripts section:

```
#Requires -Version 5.1 -Modules @{ModuleName="PSFalcon";ModuleVersion='2.1'}
param(
    [switch] $Confirm
)
try {
    # Use Find-FalconDuplicate to find duplicate hosts
    $Duplicates = Find-FalconDuplicate

    if ($Duplicates.device_id -and $Confirm) {
        # Output duplicate list to CSV
        $Duplicates | Export-Csv -Path .\duplicates.csv -NoTypeInformation

        # Use Invoke-FalconHostAction to hide hosts
        Invoke-FalconHostAction -Name hide_host -Ids $Duplicates.device_id
    } elseif ($Duplicates) {
        # Output list of duplicates
        Write-Output "Found $($Duplicates.count) potential duplicate hosts"
        Write-Output "Check ./duplicates.csv for results"
        $Duplicates | Export-Csv -Path .\duplicates.csv -NoTypeInformation
        Write-Output $Duplicates
    } else {
       Write-Output "No duplicates found."
    }
} catch {
    Write-Error $_
}
```

So far it is creating the duplicate.csv and writing out the found duplicates, but it is not hiding or removing them in our Console. The list I generated this morning was ~185 hosts and we only have 15 in our 'trash' in host management that are not from this script.

**Reply by bk-cs** [reply]
Are you running the script with the `-Confirm` switch?

**Reply by lsmithg12** [reply]
Thank you! I clearly did not look at the code right :) That has fixed my issue.

**Comment by Aphetoros** [comment]
Hello @bk-cs 

Just wanted to chime in here, we're also experiencing the same problem with `Invoke-FalconHostAction` Our code to find duplicates has always been working, is now throwing exceptions. I updated our module to the latest 2.2.2 as of this writing, same issue.

```
Import-Module -Name PSFalcon

Request-FalconToken -Cloud us-2 -ClientId '' -ClientSecret ''

Find-FalconDuplicate | Export-Csv '\Duplicates.csv'

$Duplicates = (Import-Csv '\Duplicates.csv').device_id
Invoke-FalconHostAction -Name hide_host -Id $Duplicates
```

**Reply by bk-cs** [reply]
Can you provide the exceptions with a `trace_id`?
